src/core/impl/TChem_Impl_SingleParticleAerosolWater.hpp

```diff
@@ -0,0 +1,167 @@
+
+    using const_real_type_1d_view_type = Tines::value_type_1d_view<const real_type,device_type>;
+
+    template<typename MemberType>
+    KOKKOS_INLINE_FUNCTION static
+    void team_invoke(const MemberType& member,
+        ordinal_type i_part,
+        real_type number_conc,
+        real_type state[],  // TODO: figure out how the state vector actually works
+        //const AerosolModelData amd
+        const aerosol_model_data_type& amcd,
+        ordinal_type rh_idx,
+        ordinal_type aqueous_water_idx
+        //const aerosol_ion_pair_type ionpair,
+        //const std::map<std::string, int> ion_idx_map,
+        //const std::map<std::string, real_type> atomic_weights  // TODO: pack into ionpair objects
+        )
+    {   
+        // NOTE: Cannot use std::string with cuda 
+        // Declare and initialize variables for later use
+        
+        ordinal_type anion_idx;
+        ordinal_type cation_idx;
+        ordinal_type ion_species_idx;
+        ordinal_type qty;
+        std::string atom_name;
+        std::string ion_name;
+        real_type ion_molecular_weight;
+        real_type cation;
+        real_type anion;
+        real_type e_ac;
+        real_type e_aa;
+        real_type ALPHA = -100.0;
+        real_type ion_conc;
+        real_type water_content;
+        real_type NW;
+        real_type MW;
+        real_type ZW;
+        real_type MW_H20 = 999.7351; // 55.51*18.01
+        ordinal_type ion_idx;
+
+        real_type rh = state[rh_idx];
+
+        
+        // note that you have to convert std::string to a C-string for printf since printf is a C function
+        //printf("[AerosolWater_SingleParticle::team_invoke] calculation type %s\n", ionpair.calc_type.c_str());
+
+        auto aerowater_params = amcd.aerowater_params;
+        //auto aerowater_model = amd.aerowater_model;
+
+        //int n_ionpairs = aerowater_model.ion_pair_vec.size();
+        int n_ionpairs = amcd.nAerosolWater_ionpairs; // note have to use extent instead of size since this a kokkos view
+        //printf("Number of elements in aerowater_params_: %d\n", n_ionpairs);
+        for (int j=0; j<n_ionpairs; j++){
+            //auto i_ionpair = aerowater_model.ion_pair_vec[j];
+            auto i_ionpair = aerowater_params(j);
+            //printf("[AerosolWater_SingleParticle::team_invoke] %s\n", i_ionpair.calc_type.c_str());
+            real_type effective_aw = 0.0;
+            real_type molality = 0.0;
+
+            switch( i_ionpair.calc_type ){
+                case JACOBSON:
+                    //printf("[AerosolWater_SingleParticle::team_invoke] in jacobson evaluation\n");
+                    effective_aw = max(rh, i_ionpair.jacobson_low_rh);
+
+                    // calculate molality from power series 
+                    for (int i_order=0; i_order < N_JACOBSON_COEFFS; i_order++) {
+                        molality += i_ionpair.jacobson_y_j[i_order]*std::pow(effective_aw, i_order);
```

Comment:
Change std::pow to ats<real_type>::pow. We cannot use std:: routines in src/impl/.